MapleServer2/PacketHandlers/Game/JobHandler.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.Extensions.Logging;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Servers.Game;
+using MapleServer2.Packets;
+using MapleServer2.Types;
+
+namespace MapleServer2.PacketHandlers.Game {
+    public class JobHandler : GamePacketHandler {
+        public override ushort OpCode => RecvOp.JOB;
+
+        public JobHandler(ILogger<JobHandler> logger) : base(logger) { }
+
+        public override void Handle(GameSession session, PacketReader packet) {
+            byte mode = packet.ReadByte();
+            switch (mode) {
+                case 8: // Close Skill Tree
+                    HandleCloseSkillTree(session, packet);
+                    break;
+                case 9: // Save Skill Tree
+                    HandleSaveSkillTree(session, packet);
+                    break;
+            }
+        }
+
+        public void HandleCloseSkillTree(GameSession session, PacketReader packet) {
+            session.Send(JobPacket.Close());
+        }
+
+        public void HandleSaveSkillTree(GameSession session, PacketReader packet) {
+            // Get skill tab to update
+            SkillTab skillTab = session.Player.SkillTabs[0]; // Get first skill tab only for now, uncertain of how to have multiple skill tabs
+
+            // Read skills
+            int count = packet.ReadInt(); // Number of skills
+            for (int i = 0; i < count; i++) {
+                // Read skill info
+                int id = packet.ReadInt(); // Skill id
+                short level = packet.ReadShort(); // Skill level
+                byte learned = packet.ReadByte(); // 00 if unlearned 01 if learned
```

Comment:
`learned` should be a boolean then if it's only ever going to have those two states

Reply:
In this case byte is because in each count, the packet will read that last byte and move the offset to the next Skill info structure so abool will return a true or false statement and he wants to see if is 00 or 01 but also move to the next one to keep reading.

Reply:
Again, if there are only two states, a boolean makes sense here. Also the name `learned` implies you can only ever have two states, you either learned the skill or you didn't. If there are more valid values beyond `00` and `01`, then I agree that a byte is better, but not if the only values are `00` and `01`

Reply:
That is not how the client works though. I think there is some missing knowledge here. 

The client basically sends this packet to the server and the server is handling what the client sent. You can read this particular byte as 50 always and the client will still send over the proper information the next time because how the server handles the code has nothing to do with the client (assuming you don't send over bad data back to the client). You can test this by removing all code that the server sends back to the client and see what the values are here. It should stay as 0 or 1 depending on the option. What you are saying is that the client is stateful based on what the server read which doesn't really make much sense.

The rest of the understanding is also incorrect. The reason he picked byte is probably because that byte _is_ a byte and not another data type. It's pretty fair to save a byte that's sent over by the client in a byte variable. What sharpacex is saying is that because the byte only has 2 states, it should be a boolean.

However, I don't feel like it's necessary to change this to a boolean because we don't know **for sure** if there are more states.

Reply:
Then there is still an inconsistency here, except now it's in the name. If we are going off the assumption that there could be more states than 2, then `learned` is not a good name for this. What does `learned` being equal to `5` mean then? Maybe `learnState` or something that implies "hey we think this has two states, but it could have more". I see `learned` and immediately think okay so you either learned it or not. Either way, I don't feel that strongly about this, but I do think it is still inconsistent to use a name that implies only 2 states, but give it a data type that implies more than 2 states.

Reply:
I agree with both sides here, it's a matter of deciding how we want to store data, personally, I think it should use the same type as what is being read. For instance, skill level is sometimes read as a short and sometimes as an int, but in both cases it only takes up the data of a byte. I prefer to keep skill level as a short as that is what is most commonly read, and since the data type is built off of read data, I think it should match the most common data type.